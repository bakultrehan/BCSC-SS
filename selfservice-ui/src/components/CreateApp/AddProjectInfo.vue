/** * Project information */
<template>
  <v-card class="mx-auto" style="max-width: 80%;">
    <v-card class="mx-auto">
      <v-app-bar dark color="#003366">
        <v-btn
          icon
          @click="$router.push('/projectinfo/')"
          aria-label="Back Button"
        >
          <v-icon>mdi-arrow-left</v-icon>
        </v-btn>
        <v-toolbar-title>Project information</v-toolbar-title>
        <v-spacer></v-spacer>
      </v-app-bar>
      <v-form ref="form" v-model="form">
        <v-container>
          <v-row dense>
            <v-col cols="12" md="12">
              <v-card class="pa-4 pt-6">
                <v-card-title class="headline"
                  >Project information</v-card-title
                >
                <v-card-subtitle class="text-left"
                  >Tell us about your Project</v-card-subtitle
                >
                <Input
                  v-model="organizationName"
                  counter="200"
                  label="Organization name"
                  type="text"
                  :rules="[rules.required, rules.length(2)]"
                />
                <Input
                  v-model="projectName"
                  counter="100"
                  label="Project name"
                  type="text"
                  :rules="[rules.required, rules.length(2)]"
                />
                <TextArea
                  v-model="description"
                  label="Description"
                  type="text"
                  :rules="[rules.required]"
                />
              </v-card>
            </v-col>

            <v-col cols="12">
              <v-card class="pa-4 pt-6">
                <v-card-title class="headline">Project Roles</v-card-title>
                <v-card-subtitle class="text-left"
                  >Tell us about your role in this project</v-card-subtitle
                >
                <v-radio-group v-model="myRole" row>
                  I am
                  <v-radio label="Developer" value="1"></v-radio>
                  <v-radio label="Manager" value="2"></v-radio>
                  <v-radio label="CTO" value="3"></v-radio>For this project
                </v-radio-group>
              </v-card>
            </v-col>
            <!-- <v-form ref="form" v-model="form" class="pa-4 pt-6"> -->
            <v-col cols="12" sm="6" v-if="myRole !== '2'">
              <ProjectUsers
                :userDetails="managerDetails"
                :rules="rules"
                title="Manager"
              />
            </v-col>
            <v-col cols="12" sm="6" v-if="myRole !== '3'">
              <ProjectUsers
                :userDetails="ctoDetails"
                :rules="rules"
                title="CTO"
              />
            </v-col>
            <v-col cols="12" sm="6" v-if="myRole !== '1'">
              <ProjectUsers
                :userDetails="developerDetails"
                :rules="rules"
                title="Developer"
              />
            </v-col>

            <v-col cols="12">
              <v-card flat>
                <v-divider></v-divider>
                <v-card-actions>
                  <!-- <v-btn text @click="$refs.form.reset()">Clear</v-btn> -->
                  <v-spacer></v-spacer>
                  <Button
                    :disabled="!form"
                    :loading="isLoading"
                    class="white--text"
                    color="indigo accent-4"
                    depressed
                    @click="addProjectInfo"
                    >Next</Button
                  >
                </v-card-actions>
              </v-card>
            </v-col>
          </v-row>
        </v-container>
      </v-form>
    </v-card>
  </v-card>
</template>
<script lang="ts">
import { Component, Vue, Watch, Prop } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
import { ProjectUserModel, ProjectInfoModel } from '@/models/ProjectInfoModel';
import Input from '@/Atomic/Input/Input.vue';
import TextArea from '@/Atomic/TextArea/TextArea.vue';
import Button from '@/Atomic/Button/Button.vue';
import ProjectUsers from './ProjectUsers.vue';
import validationRules from '@/config/validationRules';

const ProjectInfoModule = namespace('ProjectInfoModule');

@Component({ components: { Input, TextArea, Button, ProjectUsers } })
export default class AddProjectInfo extends Vue {
  @Prop({ default: '' })
  public id!: string;
  @Prop({ default: '' })
  public action!: string;

  @ProjectInfoModule.Getter('successStatus') public successStatus!: boolean;
  @ProjectInfoModule.Getter('errorStatus') public errorStatus!: boolean;
  @ProjectInfoModule.Action('addProjectInfo') public addProjectInfoStore!: any;
  @ProjectInfoModule.Action('loadProjectInfo')
  public loadProjectInfo!: any;
  @ProjectInfoModule.Getter('getSingleProjectInfo')
  public getSingleProjectInfo!: any;
  @ProjectInfoModule.Action('updateProjectInfo')
  public updateProjectInfoStore!: any;
  @ProjectInfoModule.Action('loadSingleProjectInfo')
  public loadSingleProjectInfo!: any;

  public form: boolean = false;
  private isLoading: boolean = false;
  private organizationName: string = '';
  private projectName: string = '';
  private description: string = '';
  private myRole: string = '1';
  private developerDetails?: undefined;
  private managerDetails?: ProjectUserModel = {
    email: '',
    phone: '',
    firstName: '',
    lastName: ''
  };
  private ctoDetails?: ProjectUserModel = {
    email: '',
    phone: '',
    firstName: '',
    lastName: ''
  };

  private isEditmode: boolean = false;
  /* istanbul ignore next */
  private rules = validationRules;

  @Watch('getSingleProjectInfo')
  private ongetSingleProjectInfoChanged(val: any) {
    this.updteEdit(val);
  }

  private addProjectInfo() {
    const data: ProjectInfoModel = {
      organizationName: this.organizationName,
      projectName: this.projectName,
      description: this.description,
      myRole: this.myRole,
      developerDetails: this.developerDetails,
      managerDetails: this.managerDetails,
      ctoDetails: this.ctoDetails
    };

    if (this.isEditmode) {
      data.id = this.id;
      this.updateProjectInfoStore(data);
    } else {
      this.addProjectInfoStore(data);
    }
    // (this.$refs.form as HTMLFormElement).reset();
    this.$router.push('/Project/technical/');
  }

  private updteEdit(val: any) {
    this.organizationName = val.organizationName;
    this.projectName = val.projectName;
    this.description = val.description;
    this.myRole = val.myRole;
    this.developerDetails = val.developerDetails;
    this.managerDetails = val.managerDetails;
    this.ctoDetails = val.ctoDetails;
    this.id = val.id;
    this.isEditmode = true;
  }

  private mounted() {
    this.isEditmode = false;
    if (this.action && this.action === 'edit' && this.id !== '') {
      this.isEditmode = true;
      this.loadSingleProjectInfo(this.id);
    }
  }
  private input(value: string) {
    // console.log('value', value);
  }
}
</script>
