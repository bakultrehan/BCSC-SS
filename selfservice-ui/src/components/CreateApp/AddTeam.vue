/* AddTeam component */
<template>
  <v-card class="mx-auto outer-card">
    <v-toolbar flat class="bc-subtitle" dark>
      <v-btn icon @click="goBack()" :aria-label="$t('team.btnBack')">
        <v-icon>mdi-arrow-left</v-icon>
      </v-btn>
      <v-toolbar-title>{{ $t('team.pageTitle') }}</v-toolbar-title>
      <div class="flex-grow-1"></div>
    </v-toolbar>
    <v-divider></v-divider>
    <v-col cols="12">
      <TeamRoles :id="id" />
    </v-col>
    <v-col cols="12">
      <v-card flat class="ma-5">
        <v-divider></v-divider>
        <v-card-actions>
          <v-spacer></v-spacer>
          <Button @click="goBack" :aria-label="$t('team.btnBack')" secondary class="back-btn">
            {{
            $t(
            showWizardExperience()
            ? 'team.btnBack'
            : 'team.btnCancel'
            )
            }}
          </Button>
          <Button class="white--text ml-6" depressed @click="nextRedirect">
            {{
            $t(
            showWizardExperience()
            ? 'team.btnNext'
            : 'team.btnSaveChanges'
            )
            }}
          </Button>
        </v-card-actions>
      </v-card>
    </v-col>
  </v-card>
</template>
<script lang="ts">
import { Component, Vue, Watch, Prop } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
import Button from '@/Atomic/Button/Button.vue';
import TeamRoles from '@/components/TeamRoles/TeamRoles.vue';
const SharedModule = namespace('SharedModule');

@Component({
  components: {
    Button,
    TeamRoles
  }
})
export default class AddTeam extends Vue {
  @Prop({ default: 0 })
  public id!: number;

  @SharedModule.Action('redirectFromSummaryPage')
  public redirectFromSummaryPage!: any;
  @SharedModule.Getter('isRedirectFromSummaryPage')
  public isRedirectFromSummaryPage!: boolean;

  private projectId: number = this.id || 0;

  private goBack() {
    const redirectPage = this.showWizardExperience() ? 'info' : 'summary';
    this.redirectFromSummaryPage(false);
    this.$router.push(`/project/${this.projectId}/${redirectPage}/`);
  }
  private showWizardExperience() {
    if (this.isRedirectFromSummaryPage) {
      return false;
    }
    return true;
  }

  private nextRedirect() {
    const redirectPage = this.showWizardExperience() ? 'technical' : 'summary';
    this.$router.push(`/project/${this.projectId}/${redirectPage}/`);
  }
}
</script>

<style lang="scss" scoped>
</style>
