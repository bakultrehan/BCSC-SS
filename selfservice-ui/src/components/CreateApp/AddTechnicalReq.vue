/** * Add TechnicalReq */
<template>
  <v-card class="mx-auto" style="max-width: 80%;">
    <!--  <v-alert type="success" v-if="successStatus">TechnicalReq {{isEditmode ? 'Updated' : 'Added'}} succesfully</v-alert>
    <v-alert type="error" v-if="errorStatus">Something went wrong...</v-alert>-->
    <v-card class="mx-auto">
      <v-app-bar dark class="bc-subtitle">
        <v-btn icon @click="$router.push(`/project/${projectId}/info/`)" aria-label="Back Button">
          <v-icon>mdi-arrow-left</v-icon>
        </v-btn>
        <v-toolbar-title>{{$t('technicalRequirements.technicalTitle')}}</v-toolbar-title>
        <v-spacer></v-spacer>
      </v-app-bar>

      <v-form ref="form" v-model="form" class="pa-4 pt-6">
        <v-container>
          <v-row dense>
            <v-col cols="12" md="12">
              <v-card class="pa-4 pt-6">
                <!-- <v-card-title class="headline"
                  >Technical information</v-card-title
                >-->
                <!-- <v-card-subtitle class="text-left"
                  >Tell us about your Project</v-card-subtitle
                >-->
                <v-card-title
                  class="headline padding-0 text-capitalize"
                >{{$t('technicalRequirements.projectName')}}{{ getSingleProjectInfo && getSingleProjectInfo.projectName }}</v-card-title>

                <Input
                  v-model="clientUri"
                  counter="500"
                  :label="$t('technicalRequirements.labelApplicationUrl')"
                  type="text"
                  :rules="[rules.required, rules.url, rules.maxLength(500)]"
                />
                <!-- <div class="col-12"> -->
                <v-card-subtitle
                  class="text-left padding-0"
                >{{$t('technicalRequirements.inputUrlText')}}</v-card-subtitle>
                <!-- </div> -->
                <div
                  v-for="(redirectUri, index) in redirectUris"
                  v-bind:key="index"
                  class="row v-form px-4"
                >
                  <v-text-field
                    v-model="redirectUris[index]"
                    :label="$t('technicalRequirements.labelRedirectUrl')"
                    type="text"
                    filled
                    @blur="addUri"
                    append-icon="mdi-minus"
                    @click:append="clearUri(index)"
                    @click:prepend="addUri"
                    :rules="[rules.url]"
                    class="addUri"
                    outlined
                  ></v-text-field>
                  <!-- :rules="[rules.required]" -->
                </div>
                <v-card-subtitle
                  class="text-left padding-0"
                >{{$t('technicalRequirements.JWKSText')}}</v-card-subtitle>
                <Input
                  v-model="jwksUri"
                  counter="500"
                  :label="$t('technicalRequirements.labelJWKSUrl')"
                  type="text"
                  :rules="[rules.required, rules.url, rules.maxLength(500)]"
                  class="pt-6"
                />
                <div class="row">
                  <div class="col-5">
                    <Select
                      v-model="idTokenSignedResponseAlg"
                      :label="$t('technicalRequirements.labelIdTokenSignedResponseAlg')"
                      :items="tokenAlgoritham"
                      :rules="[rules.required]"
                      outlined
                    />
                  </div>
                  <v-spacer />
                  <div class="col-5">
                    <Select
                      v-model="userinfoSignedResponseAlg"
                      :label="$t('technicalRequirements.labelUserinfoSignedResponseAlg')"
                      :items="userAlgoritham"
                      :rules="[rules.required]"
                      outlined
                      class="col-6"
                    />
                  </div>
                </div>
                <!-- </v-form> -->
                <v-divider></v-divider>
                <v-card-actions>
                  <!-- <v-btn text @click="$refs.form.reset()">Clear</v-btn> -->
                  <v-spacer></v-spacer>
                  <Button
                    @click="$router.push(`/project/${projectId}/info/`)"
                    aria-label="Back Button"
                    secondary
                  >{{$t('technicalRequirements.btnBack')}}</Button>
                  <Button
                    :disabled="!form"
                    :loading="isLoading"
                    class="white--text submit-req ml-6"
                    depressed
                    @click="addTechnicalReq"
                  >Next{{$t('technicalRequirements.next')}}</Button>
                </v-card-actions>
              </v-card>
            </v-col>
          </v-row>
        </v-container>
      </v-form>
    </v-card>
  </v-card>
</template>
<script lang="ts">
import { Component, Vue, Watch, Prop } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
import Input from '@/Atomic/Input/Input.vue';
import Button from '@/Atomic/Button/Button.vue';
import Select from '@/Atomic/Select/Select.vue';
import validationRules from '@/config/validationRules';

import {
  tokenSignatureAlgoritham,
  userInfoSignedResponseAlgoritham
} from '@/constants/algoritham';
import { TechnicalReqModel } from '@/models/TechnicalReqModel';
const TechnicalReqModule = namespace('TechnicalReqModule');
const ProjectInfoModule = namespace('ProjectInfoModule');

@Component({
  components: { Input, Button, Select }
})
export default class AddTechnicalReq extends Vue {
  @Prop({ default: 0 })
  public id!: number;
  @Prop({ default: '' })
  public action!: string;

  @TechnicalReqModule.Getter('successStatus') public successStatus!: boolean;
  @TechnicalReqModule.Getter('errorStatus') public errorStatus!: boolean;
  @TechnicalReqModule.Action('addTechnicalReq')
  public addTechnicalReqStore!: any;
  @TechnicalReqModule.Getter('getTechnicalReq')
  public getTechnicalReq!: any;
  @TechnicalReqModule.Action('updateTechnicalReq')
  public updateTechnicalReqStore!: any;
  @TechnicalReqModule.Action('loadTechnicalReqDetails')
  public loadTechnicalReqDetails!: any;

  @ProjectInfoModule.Getter('getSingleProjectInfo')
  public getSingleProjectInfo!: any;
  @ProjectInfoModule.Action('loadSingleProjectInfo')
  public loadSingleProjectInfo!: any;

  public form: boolean = false;
  private isLoading: boolean = false;
  private projectId: number = this.id || 0;
  private clientUri: string = '';
  private redirectUris: any = [''];
  private jwksUri: string = '';
  private idTokenSignedResponseAlg: string = 'RS256';
  private userinfoSignedResponseAlg: string = 'RS256';
  private tokenAlgoritham: any = tokenSignatureAlgoritham;
  private userAlgoritham: any = userInfoSignedResponseAlgoritham;
  private blockRemoval = true;

  // private id: string = '';
  private isEditmode: boolean = false;
  /* istanbul ignore next */
  private rules = validationRules;

  @Watch('getTechnicalReq')
  private ongetTechnicalReqChanged(val: any) {
    this.updteEdit(val);
  }

  @Watch('getSingleProjectInfo')
  private ongetSingleProjectInfoChanged(val: any) {
    this.projectId = this.getSingleProjectInfo.id;
  }

  private addTechnicalReq() {
    const data: TechnicalReqModel = {
      projectId: this.projectId,
      clientUri: this.clientUri,
      redirectUris: this.redirectUris,
      jwksUri: this.jwksUri,
      idTokenSignedResponseAlg: this.idTokenSignedResponseAlg,
      userinfoSignedResponseAlg: this.userinfoSignedResponseAlg
    };

    if (this.isEditmode) {
      data.id = this.id;
      this.updateTechnicalReqStore(data);
    } else {
      this.addTechnicalReqStore(data);
    }
    // (this.$refs.form as HTMLFormElement).reset();
    // this.$router.push('/technicalreq/');
  }

  private updteEdit(val: any) {
    this.projectId = val.projectId;
    this.clientUri = val.clientUri;
    this.redirectUris = val.redirectUris;
    this.jwksUri = val.jwksUri;
    this.idTokenSignedResponseAlg = val.idTokenSignedResponseAlg;
    this.userinfoSignedResponseAlg = val.userinfoSignedResponseAlg;
    this.id = val.id;
    this.isEditmode = true;
  }

  private mounted() {
    this.isEditmode = false;
    if (this.action && this.action === 'edit' && this.id !== 0) {
      this.isEditmode = true;
      this.loadTechnicalReqDetails(this.id);
    }

    if (this.getSingleProjectInfo && this.getSingleProjectInfo.id) {
      this.projectId = this.getSingleProjectInfo.id;
    } else {
      this.loadSingleProjectInfo(this.id);
    }
  }
  private addUri() {
    const checkEmptyLines = this.redirectUris.filter((uri: any) => uri === '');
    if (checkEmptyLines.length >= 1 && this.redirectUris.length > 0) {
      return;
    }
    this.redirectUris.push('');
  }
  private clearUri(uriId: number) {
    this.blockRemoval = this.redirectUris.length <= 1;
    if (!this.blockRemoval) {
      this.redirectUris.splice(uriId, 1);
    }
  }
}
</script>
<style lang="scss" scoped>
.padding-0 {
  padding-left: 0px !important;
}
</style>
