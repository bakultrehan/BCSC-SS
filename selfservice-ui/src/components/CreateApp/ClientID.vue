/** * ClientID component */

<template>
  <v-card class="mx-auto" style="max-width: 80%;">
    <v-toolbar flat class="bc-subtitle" dark>
      <v-btn
        icon
        @click="$router.push(`/project/${projectId}/summary/`)"
        :aria-label="$t('ClientID.Back')"
      >
        <v-icon>mdi-arrow-left</v-icon>
      </v-btn>
      <v-toolbar-title>{{ $t('ClientID.pagetitle') }}</v-toolbar-title>
      <div class="flex-grow-1"></div>
      <v-col class="col-lg-4 col-md-5 col-8">
        <v-alert type="error" v-if="errorStatus" class="alert-top"
          >Something went wrong...</v-alert
        >
      </v-col>
      <div class="flex-grow-1"></div>
    </v-toolbar>
    <v-divider></v-divider>
    <v-item-group>
      <v-container>
        <v-row class="ma-5">
          <v-snackbar v-model="snackbar" :timeout="timeout">
            {{ text }}
            <v-btn color="blue" text @click="snackbar = false">
              {{ $t('ClientID.labelSnackbarClose') }}</v-btn
            >
          </v-snackbar>

          <v-col cols="12" md="12">
            <v-card flat>
              <v-list-item-content class="headline">{{
                $t('ClientID.titleApiKey')
              }}</v-list-item-content>
              <v-list-item-content
                v-html="$t('ClientID.titleApiKeyInfo')"
              ></v-list-item-content>
            </v-card>
            <v-item class="client-id" :value="getApiData.clientId">
              <v-card
                class="d-flex align-center px-5 client-id v-card--link"
                @click="docopy(getApiData.clientId)"
              >
                <v-list-item two-line>
                  <v-list-item-action>
                    <v-list-item-title class="title mb-1">
                      {{ getApiData.clientId }}
                    </v-list-item-title>
                  </v-list-item-action>
                </v-list-item>
                <v-spacer></v-spacer>

                <div class="display-3 flex-grow-1 text-center mr-5">
                  <v-icon color="#eae9e9" large>mdi-content-copy</v-icon>
                </div>
              </v-card>
            </v-item>
          </v-col>

          <v-col cols="12" md="12">
            <v-card flat>
              <v-list-item-content class="headline">{{
                $t('ClientID.titleClientSecret')
              }}</v-list-item-content>
              <v-list-item-content
                v-html="$t('ClientID.titleClientSecretInfo')"
              ></v-list-item-content>
            </v-card>
            <v-item class="client-id" :value="getApiData.clientSecret">
              <v-card
                class="d-flex align-center px-4 client-id v-card--link"
                @click="docopy(getApiData.clientSecret)"
              >
                <v-list-item two-line>
                  <v-list-item-content>
                    <v-list-item-title class="title mb-1">
                      {{ getApiData.clientSecret }}
                    </v-list-item-title>
                  </v-list-item-content>
                </v-list-item>
                <v-spacer></v-spacer>

                <div class="display-3 flex-grow-1 text-center mr-5">
                  <v-icon color="#eae9e9" large>mdi-content-copy</v-icon>
                </div>
              </v-card>
            </v-item>
          </v-col>
          <v-col cols="12" md="12">
            <v-card flat>
              <v-list-item-content class="headline">{{
                $t('ClientID.titleTestAccount')
              }}</v-list-item-content>
              <v-list-item-content
                v-html="$t('ClientID.titleTestAccountInfo')"
              ></v-list-item-content>
            </v-card>
            <v-item class="client-id" :value="getApiData.clientSecret">
              <v-card class="d-flex align-center pa-4 client-id v-card--link">
                <v-list-item two-line>
                  <v-list-item-content>
                    <v-list-item-title
                      class="subtitle-1	 mb-1"
                      v-for="account in getApiData.testUserAccounts"
                    >
                      {{ account.userName }} , {{ account.idKey }}
                    </v-list-item-title>
                  </v-list-item-content>
                </v-list-item>
                <v-spacer></v-spacer>
              </v-card>
            </v-item>
          </v-col>
        </v-row>
      </v-container>
    </v-item-group>
  </v-card>
</template>
<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
const ClientIdModule = namespace('ClientIdModule');

@Component
export default class ClientIDDetails extends Vue {
  @Prop({ default: 0 })
  public id!: number;
  @ClientIdModule.Getter('errorStatus') public errorStatus!: boolean;
  @ClientIdModule.Action('getClientIdDetails')
  public getClientIdDetails!: any;
  @ClientIdModule.Getter('getApiData') public getApiData!: [];

  private isLoading: boolean = false;
  private projectId: number = this.id || 0;
  private snackbar: boolean = false;
  private text: string = 'Copied';
  private timeout: number = 2000;

  public docopy(message: any) {
    this.$copyText(message);
    this.snackbar = true;
  }

  private mounted() {
    this.getClientIdDetails();
  }
}
</script>

<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';
.active-bg {
  background-color: $BCgovActiveBg;
}
</style>
