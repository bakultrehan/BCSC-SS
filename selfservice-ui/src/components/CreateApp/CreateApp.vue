/** * Footer of app */

<template>
  <div>Create App</div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import Input from '@/Atomic/Input/Input.vue';

@Component({
  components: {
    Input
  }
})
export default class Footer extends Vue {
  private e6: number = 0;
}
</script>

<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';
</style>
