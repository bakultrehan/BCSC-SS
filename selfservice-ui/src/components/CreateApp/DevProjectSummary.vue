/** * DevProjectSummary component */

<template>
  <v-card class="mx-auto outer-card">
    <v-toolbar flat class="bc-subtitle padding-0" dark>
      <v-btn icon @click="goBack()" :aria-label="$t('summaryPage.goBack')">
        <v-icon>mdi-arrow-left</v-icon>
      </v-btn>
      <v-toolbar-title>{{ $t('summaryPage.pagetitle') }}</v-toolbar-title>
      <div class="flex-grow-1"></div>

      <div class="flex-grow-1"></div>
    </v-toolbar>
    <v-container>
      <v-row>
        <v-col cols="8" flat>
          <v-card flat>
            <v-list-item-content
              class="text-left padding-0"
              v-html="$t('summaryPage.subTitle')"
            ></v-list-item-content>
          </v-card>
        </v-col>
        <v-col cols="4"><ProjectActions :id="id" /> </v-col>
        <ProjectSummary :id="id" />
      </v-row>
    </v-container>
  </v-card>
</template>
<script lang="ts">
import { Component, Vue, Watch, Prop } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
import ProjectSummary from '@/components/CreateApp/ProjectSummary.vue';
import ProjectActions from '@/components/ProjectActions/ProjectActions.vue';

const SharedModule = namespace('SharedModule');
const ProjectInfoModule = namespace('ProjectInfoModule');

@Component({
  components: {
    ProjectSummary,
    ProjectActions,
  },
})
export default class DevProjectSummary extends Vue {
  @Prop({ default: 0 })
  public id!: number;

  @SharedModule.Action('redirectFromSummaryPage')
  public redirectFromSummaryPage!: any;
  @ProjectInfoModule.Getter('getSingleProjectInfo')
  public getSingleProjectInfo!: any;

  private projectId: number = this.id || 0;

  private mounted() {
    this.redirectFromSummaryPage(true);
  }

  private goBack() {
    this.redirectFromSummaryPage(false);
    this.$router.push(`/project/${this.projectId}/test-account`);
  }
}
</script>

<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';
.active-bg {
  background-color: $BCgovActiveBg;
}
.card-width {
  max-width: 110px;
}
.text-center {
  text-align: center !important;
}
</style>
