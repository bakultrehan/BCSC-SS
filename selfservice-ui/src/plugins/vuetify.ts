import Vue from 'vue';
import Vuetify from 'vuetify/lib';

Vue.use(Vuetify);

export default new Vuetify({
  icons: {
    iconfont: 'mdi',
  },
  theme: {
    themes: {
      light: {
        error: '#D8292F',
        success: '#2E8540',
        warning: '#FFC107',
      },
    },
    options: {
      cspNonce: 'cLv1G2y8KtMdR',
    },
  },
});
