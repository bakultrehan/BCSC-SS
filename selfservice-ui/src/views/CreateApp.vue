<template>
  <div class="home">
    <CreateApp />
  </div>
</template>

<script>
// @ is an alias to /src
import { Component, Vue } from 'vue-property-decorator';
import CreateApp from '@/components/CreateApp/CreateApp.vue';

@Component({
  components: {
    CreateApp
  }
})
// Home goes here
export default class Home extends Vue {}
</script>
