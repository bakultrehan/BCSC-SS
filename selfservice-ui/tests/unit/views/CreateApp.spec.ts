import { shallowMount, createLocalVue } from '@vue/test-utils';
import CreateApp from '@/views/CreateApp.vue';
import vuetify from 'vuetify';

describe('CreateApp.vue', () => {
  let wrapper: any;
  beforeEach(() => {
    const localVue = createLocalVue();
    localVue.use(vuetify);

    wrapper = shallowMount(CreateApp, {
      localVue
    });
  });

  it('renders props when passed', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
