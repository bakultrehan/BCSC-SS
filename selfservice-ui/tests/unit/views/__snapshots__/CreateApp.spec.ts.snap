// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateApp.vue renders props when passed 1`] = `
<div
  class="home"
>
  <createapp-stub />
</div>
`;
